apps/ejabberd/test/commands_backend_SUITE.erl

```diff
@@ -0,0 +1,681 @@
+    ok.
+
+%% clients command funs
+get_simple_client_command(_Caller, _SomeBinary) ->
+    <<"client bob is OK">>.
+
+get_two_args_client_command(_Caller, _SomeBinary, _SomeInteger) ->
+    <<"client2 bob is OK">>.
+
+post_simple_client_command(_Caller, _Title, _Content) ->
+    <<"new_resource">>.
+
+put_simple_client_command(_Username, _Password) ->
+    changed.
+
+delete_simple_client_command(_Username, _BikeName) ->
+    changed.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%% utilities
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+build_path_prefix() ->
+    "http://" ++ ?HOST ++ ":" ++ integer_to_list(?PORT).
+
+maybe_add_body([]) ->
+    [];
+maybe_add_body(Args) ->
+    jiffy:encode(maps:from_list(Args)).
+
+maybe_add_accepted_headers("POST") ->
+    accepted_headers();
+maybe_add_accepted_headers("PUT") ->
+    accepted_headers();
+maybe_add_accepted_headers(_) ->
+    [].
+
+accepted_headers() ->
+    [{<<"Content-Type">>, <<"application/json">>}, {<<"Accept">>, <<"application/json">>}].
+
+maybe_add_auth_header({User, Password}) ->
+    Basic = list_to_binary("basic " ++ base64:encode_to_string(User ++ ":"++ Password)),
+    [{<<"authorization">>, Basic}];
+maybe_add_auth_header(admin) ->
+    [].
+
+-spec create_path_with_binds(string(), list()) -> binary().
+create_path_with_binds(Base, ArgList) when is_list(ArgList) ->
+    list_to_binary(
+        lists:flatten(Base ++ ["/" ++ to_list(ArgName) ++ "/" ++ to_list(ArgValue)
+                               || {ArgName, ArgValue} <- ArgList])).
+
+to_list(Int) when is_integer(Int) ->
+    integer_to_list(Int);
+to_list(Float) when is_float(Float) ->
+    float_to_list(Float);
+to_list(Bin) when is_binary(Bin) ->
+    binary_to_list(Bin);
+to_list(Atom) when is_atom(Atom) ->
+    atom_to_list(Atom);
+to_list(Other) ->
+    Other.
+
+-spec request(binary(), method(), admin | {{binary(), binary()}, boolean()}) -> any.
+request(Path, "GET", Entity) ->
+    request(Path, "GET", [], Entity);
+request(Path, "DELETE", Entity) ->
+    request(Path, "DELETE", [], Entity).
+
+-spec request(binary(), method(), list({atom(), any()}),
+              {headers, list()} | admin | {{binary(), binary()}, boolean()}) -> any.
+request(Path, Method, Body, {headers, Headers}) ->
+    {ok, Pid} = fusco:start_link("http://"++ ?HOST ++ ":" ++ integer_to_list(?PORT), []),
+    R = fusco:request(Pid, Path, Method, Headers, Body, 5000),
+    fusco:disconnect(Pid),
+    teardown(),
+    R;
+request(Path, Method, BodyData, {{_User, _Pass} = Auth, Authorized}) ->
+    setup(client_module()),
+    meck:expect(ejabberd_auth, check_password, fun(_, _, _) -> Authorized end),
+    Body = maybe_add_body(BodyData),
+    AuthHeader = maybe_add_auth_header(Auth),
+    AcceptHeader = maybe_add_accepted_headers(Method),
+    request(Path, Method, Body, {headers, AuthHeader ++ AcceptHeader});
```

Comment:
IMO, it would be much more readable if this function clause was named differently. 2 other clauses build `headers` and call the function again with only last arg changed.